### PR TITLE
Add serve command (Follow up delegate to backend) 

### PR DESCRIPTION
**Pull request by gtardif**
**- What I did**
* Add serve command delegating to backend the startup of gRPC API for context backend commands (ACI/ECS), today provided by docker/compose-cli, and started in Docker Desktop

This is based on #2922 , only the last commit is really part of this PR. (other commits will potentially get rebased from #2922 )

Fixes https://github.com/docker/dev-tooling-team/issues/253

**- How I did it**

**- How to verify it**
* place compose-cli v1.0.6 binary in /usr/local/lib/docker/cli-backends/
* `docker serve --help` displays help for serve command

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: cli/command/serve/serve.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package serve
+
+import (
+	"github.com/docker/cli/backends"
+	"github.com/spf13/cobra"
+)
+
+// NewServeCommand command to serve gRPC API for backend commands (ACI/ECS for now) delegated to backend
```
Comment (AkihiroSuda) [inline cli/command/serve/serve.go:8]: I think this text should be also added to cobra.Command.Short.

File: cli/command/serve/serve.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+func NewServeCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                "serve",
+		Short:              "Start a Docker client api server",
```
Comment (AkihiroSuda) [inline cli/command/serve/serve.go:12]: This text sounds like starting Docker REST API server. WDYT about adding the godoc comment above here?

File: cli/command/commands/commands.go [hunk 3]
```diff
@@ -65,6 +70,9 @@ func AddCommands(cmd *cobra.Command, dockerCli command.Cli) {
 		// secret
 		secret.NewSecretCommand(dockerCli),
 
+		// service
```
Comment (AkihiroSuda) [inline cli/command/commands/commands.go:73]: Not service

**Comment by gtardif** [comment]
Closing this, will reopen/fetch code later on if required

**Pull request closed without merging**